### PR TITLE
feat(TCOMP-276): add endpoints search by tags

### PR DESCRIPTION
**Pull request by chmyga**
* Add common tags

* Add example of tags usage

**What is the current behavior?** (You can also link to an open issue here)

**What is the new behavior?**
Implementation of tags support for component definitions/wizards/datastores/datasets added in this PR. Common tags introduced to be used in TCOMP components. Endpoint is added to provide component definitions/datastores search by tags.

**Please check if the PR fulfills these requirements**

- [x] The commit message follows Talend standard
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features) ?
- [ ] The code coverage on new code >75%
- [ ] The new code does not introduce new technical issues (sonar / eslint)

**What kind of change does this PR introduce?**

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build / CI related changes
- [ ] Other... Please describe:

**Does this PR introduce a breaking change?**

- [ ] Yes
- [x] No

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...

**Other information**:

### CODE REVIEW

**Comment by build-talend-common** [comment]
Pull request test result: [failure](https://ci-common.datapwn.com/job/components_pull-request_master/168/) [:x:](https://ci-common.datapwn.com/job/components_pull-request_master/168/)

File: core/components-api/src/main/java/org/talend/components/api/CommonTags.java [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+// This source code is available under agreement available at
+// %InstallDIR%\features\org.talend.rcp.branding.%PRODUCTNAME%\%PRODUCTNAME%license.txt
+//
+// You should have received a copy of the agreement
+// along with this program; if not, write to Talend SA
+// 9 rue Pages 92150 Suresnes, France
+//
+// ============================================================================
+package org.talend.components.api;
+
```
Comment (francisu) [inline core/components-api/src/main/java/org/talend/components/api/CommonTags.java]: I think this should be in components common rather than api.

**Comment by build-talend-common** [comment]
Pull request test result: [failure](https://ci-common.datapwn.com/job/components_pull-request_master/223/) [:x:](https://ci-common.datapwn.com/job/components_pull-request_master/223/)

**Comment by build-talend-common** [comment]
Pull request test result: [success](https://ci-common.datapwn.com/job/components_pull-request_master/225/) [:star:](https://ci-common.datapwn.com/job/components_pull-request_master/225/)

**Comment by build-talend-common** [comment]
Pull request test result: [success](https://ci-common.datapwn.com/job/components_pull-request_master/231/) [:star:](https://ci-common.datapwn.com/job/components_pull-request_master/231/)

**Comment by build-talend-common** [comment]
Pull request test result: [failure](https://ci-common.datapwn.com/job/components_pull-request_master/248/) [:x:](https://ci-common.datapwn.com/job/components_pull-request_master/248/)

**Comment by build-talend-common** [comment]
Pull request test result: [success](https://ci-common.datapwn.com/job/components_pull-request_master/249/) [:star:](https://ci-common.datapwn.com/job/components_pull-request_master/249/)

File: services/components-api-service-rest/src/main/java/org/talend/components/service/rest/DefinitionsController.java [hunk 2]
```diff
@@ -51,6 +51,17 @@
...
+    /**
+     * Return all known definitions that match the given type and has the given tag.
+     *
+     * @param type the wanted definition type.
+     * @param tag the wanted tag.
+     * @return the list of all definitions that match the given type and has the given tag.
+     * @returnWrapped java.lang.Iterable<org.talend.components.service.rest.dto.DefinitionDTO>
+     */
+    @RequestMapping(value = "/definitions/{type}/{tag}", method = GET)
+    List<DefinitionDTO> listComponentDefinitionsByTag(@PathVariable("type") DefinitionType type,
```
Comment (sgandon) [inline services/components-api-service-rest/src/main/java/org/talend/components/service/rest/DefinitionsController.java]: I really would advise you add an optional parameter to the route that queries definitions of a given type.

**Comment by build-talend-common** [comment]
Pull request test result: [success](https://ci-common.datapwn.com/job/components_pull-request_master/254/) [:star:](https://ci-common.datapwn.com/job/components_pull-request_master/254/)

**Pull request merged**